### PR TITLE
Fixed issue #775. Mark the ConvertibleFromAny constructor as explicit, and fix operator overload issue for Unprintable

### PR DESCRIPTION
**Pull request by mazong1123**
I've seen the fix somewhere a couple of weeks ago, and I failed to find the source again. I'm actually not the author of this PR, someone else has already provided the exactly same solution somewhere.

Anyway, this PR fixes the issue #775 somehow.

### CODE REVIEW

File: googlemock/test/gmock-matchers_test.cc [hunk 1]
```diff
@@ -867,11 +867,12 @@ class Unprintable {
  public:
   Unprintable() : c_('a') {}
 
-  bool operator==(const Unprintable& /* rhs */) { return true; }
  private:
   char c_;
 };
 
+inline bool operator==(const Unprintable&, /* lhs */ const Unprintable& /* rhs */) { return true; }
```
Comment (BillyDonahue) [inline googlemock/test/gmock-matchers_test.cc]:
```
inline bool operator==(const Unprintable& /* lhs */,
                       const Unprintable& /* rhs */) {
  return true;
}
```

Comment (mazong1123) [inline googlemock/test/gmock-matchers_test.cc]: Should I reformat the code style as your note and submit a new PR again?

**Comment by mazong1123** [comment]
@BillyDonahue just formatted the operator== method coding style.

PS: The appveryor build for MSVC 14 always failed. But that should be caused by another issue. Please check if this patch fix #775  :)

**Comment by BillyDonahue** [comment]
Looks good, thanks!

**Pull request merged**